hyperspyui/plugins/alignzlp.py

```diff
@@ -24,4 +25,22 @@ def create_toolbars(self):
     def default(self):
         ui = self.ui
         s = ui.get_selected_signal()
-        s.align_zero_loss_peak()
+        logger.debug('Align zero loss peak of ' + str(s))
+        title = s.metadata.General.title
+
+        signal_list = [sig for sig in ui.signals if sig.signal is not s]
+        picker = SignalList(items=signal_list, parent=ui, multiselect=True)
+        diag = ui.show_okcancel_dialog("Select signals to align with {}".format(
+            title), picker, modal=True)
+        signals = []
+        if diag.result() == QDialog.Accepted:
+            signals = picker.get_selected()
+        else:
+            return
+        signals = [] if not signals else signals  # if none selected
+        signals = [sig.signal for sig in signals]
+        logger.debug('Also aligning the following signals\n' + str(signals))
+        # unclear how to incorporate this. It needs a blist argument
+        # picker.unbind(blist=?)
```

Comment:
If the `SignalList` was initialized with `ui.signals` (a `BindingList`), it would have kept the signal list in sync with the global signal list (adding/removing signals as appropriate). If that feature is used, you have to unbind afterwards. I'm sure there's a cleaner pattern somewhere involving context managers...

Reply:
To be clear, since you are creating a copy as a plain list, you can drop the unbind step.